cluster-controller/src/main/java/io/enmasse/barnabas/controller/cluster/resources/AbstractCluster.java

```diff
@@ -0,0 +1,311 @@
+package io.enmasse.barnabas.controller.cluster.resources;
+
+import io.fabric8.kubernetes.api.model.*;
+import io.fabric8.kubernetes.api.model.extensions.Deployment;
+import io.fabric8.kubernetes.api.model.extensions.DeploymentBuilder;
+import io.fabric8.kubernetes.api.model.extensions.DeploymentStrategyBuilder;
+import io.fabric8.kubernetes.api.model.extensions.RollingUpdateDeploymentBuilder;
+import io.fabric8.kubernetes.api.model.extensions.StatefulSet;
+import io.fabric8.kubernetes.api.model.extensions.StatefulSetBuilder;
+import io.fabric8.kubernetes.api.model.extensions.StatefulSetUpdateStrategyBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public abstract class AbstractCluster implements Cluster {
+
+    protected final Logger log = LoggerFactory.getLogger(getClass());
+
+    protected final String name;
+    protected final String namespace;
+    protected Map<String, String> labels = new HashMap<>();
+
+    // Docker image configuration
+    protected String image;
+    // Number of replicas
+    protected int replicas;
+
+    protected String healthCheckPath;
+    protected int healthCheckTimeout;
+    protected int healthCheckInitialDelay;
+
+    protected String headlessName;
+
+    protected AbstractCluster(String namespace, String name) {
+        this.name = name;
+        this.namespace = namespace;
+    }
+
+    public Map<String, String> getLabels() {
+        return labels;
+    }
+
+    public void setLabels(Map<String, String> newLabels) {
+        newLabels.put("cluster-name", name);
+        this.labels = new HashMap<>(newLabels);
+    }
+
+    public int getReplicas() {
+        return replicas;
+    }
+
+    public void setReplicas(int replicas) {
+        this.replicas = replicas;
+    }
+
+    public void setImage(String image) {
```

Comment:
Should all the setters be public? Wouln$t it be better to use something like protected? I think that currently they are not used from outside.

Reply:
Yes, better having them `protected` even if something like that works as well (without the need for setters)
In the static method `fromStatefulSet` in `KafkaCluster` we can just do : 

```
kafka.image = ss.getSpec().getTemplate().getSpec().getContainers().get(0).getImage();
```

instead of the current : 

```
kafka.setImage(ss.getSpec().getTemplate().getSpec().getContainers().get(0).getImage());
```